File: src/main/java/com/github/unchama/seichiassist/data/player/PlayerNickName.kt
package com.github.unchama.seichiassist.data.player

data class PlayerNickName(val style: Style = Style.SecondaryName, val id1: Int = 0, val id2: Int = 0, val id3: Int = 0) {
  enum class Style(val displayLevel: Boolean) {
    Level(true),
    SecondaryName(false),
    ;

    companion object {
      fun marshal(isLevel: Boolean): Style {
        return if (isLevel) {
          Level
        } else {
          SecondaryName
        }
      }
    }
  }
}

File: src/main/scala/com/github/unchama/seichiassist/data/player/PlayerNickName.scala
package com.github.unchama.seichiassist.data.player

sealed abstract class Style(val displayLevel: Boolean) {}

case class PlayerNickName(style: Style = PlayerNickName.Style.TitleCombination,
                          id1: Int = 0,
                          id2: Int = 0,
                          id3: Int = 0)

object PlayerNickName {

  object Style {

    def marshal(isLevel: Boolean): Style = if (isLevel) PlayerNickName.Style.Level else PlayerNickName.Style.TitleCombination

    case object Level extends Style(true)

    case object TitleCombination extends Style(false)

  }

}
